include/openrave/units.h

```diff
@@ -418,6 +432,68 @@ inline T GetAngleUnitConversionScale(const AngleUnit sourceAngleUnit, const Angl
     return GetAngleUnitStandardValue<T>(targetAngleUnit) / GetAngleUnitStandardValue<T>(sourceAngleUnit);
 }
 
+//
+// Time Stamp
+//
+
+// \brief how many units in a second
+template <typename T>
+inline T GetStampUnitStandardValue(const char* pUnit)
+{
+    // ordered from most likely
+    if (strcmp(pUnit, "s") == 0 ) {
+        return T(1.0);
+    }
+    if (strcmp(pUnit, "ms") == 0 ) {
+        return T(1e3);
+    }
+    if (strcmp(pUnit, "us") == 0 ) {
+        return T(1e6);
+    }
+    throw OPENRAVE_EXCEPTION_FORMAT("Unsupported time stamp unit '%s'", pUnit, ORE_StampUnitInvalid);
+}
+
+template <typename T>
+inline T GetStampUnitStandardValue(const std::string& s) {
+    return GetStampUnitStandardValue<T>(s.c_str());
+}
+
+// \brief how many units in a second
+template <typename T>
+constexpr inline T GetStampUnitStandardValue(const StampUnit unit)
+{
+    if( unit == OpenRAVE::SU_Second ) {
+        return T(1.0);
+    }
+    if( unit == OpenRAVE::SU_Millisecond ) {
+        return T(1e3);
+    }
+    if( unit == OpenRAVE::SU_Microsecond ) {
+        return T(1e6);
+    }
+
+    throw OPENRAVE_EXCEPTION_FORMAT("Unsupported time stamp unit '%s'", GetStampUnitString(unit), ORE_StampUnitInvalid);
+    return T{};
+}
+
+template <typename T>
+inline T GetStampUnitConversionScale(const std::string &sourceStampUnit, const std::string &targetStampUnit)
```

Comment:
I think TimeStamp units conversation has to be handled differently since we could have integers, and I don't want to introduce floating point...